controllers/release/adapter_test.go

```diff
@@ -656,58 +432,150 @@ var _ = Describe("Release Adapter", Ordered, func() {
 			adapter = createReleaseAndAdapter()
+		})
 
-			var err error
-			pipelineRun, err = adapter.createReleasePipelineRun(releaseStrategy, enterpriseContractPolicy, snapshot)
-			Expect(pipelineRun).NotTo(BeNil())
+		It("should stop processing if the release has finished", func() {
+			adapter.release.MarkReleasing("")
+			adapter.release.MarkReleased()
+
+			result, err := adapter.EnsureReleaseIsRunning()
+			Expect(!result.RequeueRequest && result.CancelRequest).To(BeTrue())
 			Expect(err).NotTo(HaveOccurred())
 		})
 
-		It("returns a PipelineRun", func() {
-			Expect(reflect.TypeOf(pipelineRun)).To(Equal(reflect.TypeOf(&v1beta1.PipelineRun{})))
+		It("should mark the Release as releasing if it is missing the status", func() {
+			result, err := adapter.EnsureReleaseIsRunning()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsReleasing()).To(BeTrue())
 		})
 
-		It("has owner annotations", func() {
-			Expect(pipelineRun.GetAnnotations()[handler.NamespacedNameAnnotation]).To(ContainSubstring(adapter.release.Name))
-			Expect(pipelineRun.GetAnnotations()[handler.TypeAnnotation]).To(ContainSubstring("Release"))
-		})
+		It("should do nothing if the release is already running", func() {
+			adapter.release.MarkReleasing("")
 
-		It("has release labels", func() {
-			Expect(pipelineRun.GetLabels()[tekton.PipelinesTypeLabel]).To(Equal("release"))
-			Expect(pipelineRun.GetLabels()[tekton.ReleaseNameLabel]).To(Equal(adapter.release.Name))
-			Expect(pipelineRun.GetLabels()[tekton.ReleaseNamespaceLabel]).To(Equal(testNamespace))
+			result, err := adapter.EnsureReleaseIsRunning()
+			Expect(!result.RequeueRequest && !result.CancelRequest).To(BeTrue())
+			Expect(err).NotTo(HaveOccurred())
+			Expect(adapter.release.IsReleasing()).To(BeTrue())
 		})
+	})
 
-		It("references the pipeline specified in the ReleaseStrategy", func() {
-			Expect(pipelineRun.Spec.PipelineRef.Name).To(Equal(releaseStrategy.Spec.Pipeline))
+	Context("When EnsureReleaseIsProcessed is called", func() {
+		var adapter *Adapter
+
+		AfterEach(func() {
+			_ = adapter.client.Delete(ctx, adapter.release)
 		})
 
-		It("contains a parameter with the json representation of the EnterpriseContractPolicy", func() {
-			jsonSpec, _ := json.Marshal(enterpriseContractPolicy.Spec)
-			Expect(pipelineRun.Spec.Params).Should(ContainElement(HaveField("Value.StringVal", Equal(string(jsonSpec)))))
+		BeforeEach(func() {
+			adapter = createReleaseAndAdapter()
 		})
 
-		It("contains a parameter with the json representation of the Snapshot", func() {
-			jsonSpec, _ := json.Marshal(snapshot.Spec)
-			Expect(pipelineRun.Spec.Params).Should(ContainElement(HaveField("Value.StringVal", Equal(string(jsonSpec)))))
+		It("should do nothing if the Release is already processed", func() {
+			adapter.release.MarkProcessing("")
+			adapter.release.MarkProcessed()
+
+			result, err := adapter.EnsureReleaseIsRunning()
```

Comment:
shouldn't this be `EnsureReleaseIsProcessed`?

Reply:
Good catch.